src/nxt_fs.c

```diff
@@ -62,11 +62,14 @@ nxt_fs_mkdir_parent(const u_char *path, mode_t mode)
     ret = NXT_OK;
 
     ptr = strrchr(dir, '/');
-    if (nxt_fast_path(ptr != NULL)) {
-        *ptr = '\0';
-        ret = nxt_fs_mkdir((const u_char *) dir, mode);
+    if (nxt_slow_path(ptr == NULL)) {
+        goto out_free;
     }
 
+    *ptr = '\0';
+    ret = nxt_fs_mkdir((const u_char *) dir, mode);
+
+out_free:
     nxt_free(dir);
 
     return ret;
```

Comment:
I'm not sure that's an improvement. It removes a level of indentation, but adds _more_ code...

Reply:
Yeah, it's not an improvement per se, but when you add the following commits, which add more levels of nesting, I think it pays off, IMO.

Reply:
Maybe note that in the commit message...

Reply:
ACK